### PR TITLE
Modify the default port

### PR DESCRIPTION
**Pull request by PenghaiZhang**
* Set the default proxy port empty
* Allow the value of port to be empty

### CODE REVIEW

**Comment by PenghaiZhang** [comment]
Hi @abidingotter , could you please have a look again as we found a new bug?
The adminconsole JAR did not get running with the bundled JRE, which means clients who have no JRE installed  can only run the launcher.

File: src/main/java/org/apereo/openequella/adminconsole/service/JarService.java [hunk 1]
```diff
@@ -136,7 +136,11 @@ public void downloadJar(String jarName) {
 	public int executeJar(String jarName, String mainClass, String... jvmArgs) {
 		final File binJar = getBinJarFile(jarName);
 		final List<String> command = new ArrayList<>();
-		command.add("java");
+		// Use the bundled JRE to run admin console
+		final String os = System.getProperty("os.name");
+		final String bundledJrePath = System.getProperty("user.dir")+"/jdk8u212-b03-jre/";
```
Comment (abidingotter) [inline src/main/java/org/apereo/openequella/adminconsole/service/JarService.java]: Could you change to:
```
File currentPath = new File(System.getProperty("user.dir"));
File bundledJrePath = new File(currentPath, "jdk8u212-b03-jre");
File binPath = new File(bundledJrePath, os.toLowerCase().indexOf("mac") > 0 ? "Contents/Home/bin/" : "bin");
String fullJavaPath = new File(binPath, "java").getAbsolutePath();
```
It's a bit more wordy, but you don't have to deal with \ or / path separators

File: src/main/java/org/apereo/openequella/adminconsole/service/JarService.java [hunk 2]
```diff
@@ -136,7 +136,13 @@ public void downloadJar(String jarName) {
 	public int executeJar(String jarName, String mainClass, String... jvmArgs) {
 		final File binJar = getBinJarFile(jarName);
 		final List<String> command = new ArrayList<>();
-		command.add("java");
+		// Use the bundled JRE to run admin console
+		final String os = System.getProperty("os.name");
```
Comment (abidingotter) [inline src/main/java/org/apereo/openequella/adminconsole/service/JarService.java]: Hmm, actually, you can't run this from the debugger anymore.  I will try to think of a solution

**Comment by PenghaiZhang** [comment]
As I requested Aaron to review and he also made some changes, so need approval from Ian.

**Pull request merged**